tools/spdx-check.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+BASE_COMMIT=$1
+
+# List of files to check, excluding vendor directories
+files=$(git diff --name-only --diff-filter=A "${BASE_COMMIT}"..HEAD | grep -v "vendor/")
+
+# SPDX License Identifier to check for
+license_identifier="SPDX-License-Identifier:"
+
+# Array of file extensions to check for SPDX header
+file_extensions=("sh" "c" "go" "py" "rs" "yaml" "yml")
+file_names=("Dockerfile" "Makefile")
```

Comment:
Shouldn't we check Dockerfile.in as well?

Reply:
Will add. Any other suggestions? *.mk?

Reply:
I don't think so. Regarding .mk files, our kernel .mk files are minimal and the get-deps.mk is auto-generated, so I think it's fine, we don't need to check license on those...

Reply:
Yeah, I've just added the Dockerfile.in one and also the .h extension.